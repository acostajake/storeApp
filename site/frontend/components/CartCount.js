import PropTypes from 'prop-types';
import React from 'react';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import styled from 'styled-components';
import CartItem from './CartItem';

const AnimatedStyles = styled.span`
	position: relative;
	.count {
		display: block;
		position: relative;
		transition: all 0.6s;
		backface-visibility: hidden;
	}
	.count-enter {
		transform: rotateX(0.5turn);
	}
	.count-enter-active {
		transform: rotateX(0);
	}
	.count-exit {
		top: 0;
		position: absolute;
		transform: rotateX(0);
	}
	.count-exit-active {
		transform: rotateX(0.5turn);
	}
`;

const Dot = styled.div`
	background: ${(props) => props.theme.darkblue};
	color: white;
	border-radius: 50%;
	padding: 0.5rem;
	margin-left: 0.5rem;
	height: 3rem;
	min-width: 3rem;
	font-weight: 100;
	font-feature-settings: 'tnum';
	font-variant-numeric: tabular-nums;
`;

const CartCount = ({ count }) => (
	<AnimatedStyles>
		<TransitionGroup>
			<CSSTransition
				unmountOnExit
				className='count'
				classNames='count'
				key={count}
				timeout={{ enter: 600, exit: 600 }}
			>
				<Dot>{count}</Dot>
			</CSSTransition>
		</TransitionGroup>
	</AnimatedStyles>
);

CartCount.propTypes = {
	count: PropTypes.number.isRequired,
};

export default CartCount;
