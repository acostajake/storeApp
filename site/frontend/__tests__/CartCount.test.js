import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import CartCount from '../components/CartCount';

const shallowCount = shallow(<CartCount count={4} />);

describe('<CartCount/>', () => {
	it('matches snapshot', () => {
		expect(toJSON(shallowCount)).toMatchSnapshot();
	});
	it('updates via props', () => {
		expect(toJSON(shallowCount)).toMatchSnapshot();
		shallowCount.setProps({ count: 7 });
		expect(toJSON(shallowCount)).toMatchSnapshot();
	});
});
