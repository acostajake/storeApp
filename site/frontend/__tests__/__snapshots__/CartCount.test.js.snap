// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> matches snapshot 1`] = `
<CartCount__AnimatedStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="4"
      timeout={
        Object {
          "enter": 600,
          "exit": 600,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        4
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimatedStyles>
`;

exports[`<CartCount/> updates via props 1`] = `
<CartCount__AnimatedStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="4"
      timeout={
        Object {
          "enter": 600,
          "exit": 600,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        4
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimatedStyles>
`;

exports[`<CartCount/> updates via props 2`] = `
<CartCount__AnimatedStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="7"
      timeout={
        Object {
          "enter": 600,
          "exit": 600,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        7
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimatedStyles>
`;
